crates/e2e/src/subxt_client.rs

```diff
@@ -444,43 +332,125 @@ where
             events: tx_events,
         })
     }
+}
 
-    /// Executes a `call` for the contract at `account_id`.
-    ///
-    /// Returns when the transaction is included in a block. The return value
-    /// contains all events that are associated with this transaction.
-    pub async fn call<Args, RetType>(
+#[async_trait]
+impl<C, E> ChainBackend for Client<C, E>
+where
+    C: subxt::Config + Send + Sync,
+    C::AccountId: Clone
+        + Debug
+        + Send
+        + Sync
+        + core::fmt::Display
+        + scale::Codec
+        + From<sr25519::PublicKey>
+        + serde::de::DeserializeOwned,
+    C::Address: From<sr25519::PublicKey>,
+    C::Signature: From<sr25519::Signature>,
+    C::Address: Send + Sync,
+    <C::ExtrinsicParams as ExtrinsicParams<C::Hash>>::OtherParams: Default + Send + Sync,
+
+    E: Environment,
+    E::AccountId: Debug + Send + Sync,
+    E::Balance: Clone
+        + Debug
+        + Send
+        + Sync
+        + TryFrom<u128>
+        + scale::HasCompact
+        + serde::Serialize,
```

Comment:
I _think_ this can be simplified (but I haven't compiled for this particular case) to:
```rust
trait EnvT: Environment { }
impl<T> EnvT for T where
	T: Environment,
	T::AccountId: Debug + Send + Sync,
	T::Balance: Clone + Debug + Sync + ... { }
```
and here, simplify to
```rust
E: EnvT
```
🤔 
(similarly for others as well)